From fa8ea1f701eb73c50067e0907fbaa62c54d88582 Mon Sep 17 00:00:00 2001
From: Edward Welbourne <edward.welbourne@qt.io>
Date: Mon, 29 Sep 2025 14:20:35 +0200
Subject: [PATCH 253/553] tst_QDateTimeEdit::stepModifierButtons(): make format
 overt

The test's data rely on getting HH:mm:ss time format, so make that
explicit in the test, rather than taking it for granted. It's been
working until now only because the C locale uses that as its short
format, which is about to change.

Task-number: QTBUG-140643
Pick-to: 6.8 6.5
Change-Id: Icf8c700740872c6e8805aa612b2a14beec8fb44a
Reviewed-by: Mitch Curtis <mitch.curtis@qt.io>
(cherry picked from commit c1366c357bcec1e449fc71f23a4d991706d184d6)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 tests/auto/widgets/widgets/qdatetimeedit/tst_qdatetimeedit.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/auto/widgets/widgets/qdatetimeedit/tst_qdatetimeedit.cpp b/tests/auto/widgets/widgets/qdatetimeedit/tst_qdatetimeedit.cpp
index f3d6d9c78c1..2062ec91ddb 100644
--- a/tests/auto/widgets/widgets/qdatetimeedit/tst_qdatetimeedit.cpp
+++ b/tests/auto/widgets/widgets/qdatetimeedit/tst_qdatetimeedit.cpp
@@ -4419,6 +4419,7 @@ void tst_QDateTimeEdit::stepModifierButtons()
     testWidget->hide();
 
     EditorDateEdit edit;
+    edit.setDisplayFormat("HH:mm:ss");
     edit.setTime(startTime);
     edit.show();
     QVERIFY(QTest::qWaitForWindowActive(&edit));
-- 
2.51.2

